linkerd/app/core/src/proxy/server.rs

```diff
@@ -164,7 +170,11 @@ where
                     .clone()
                     .into_service()
                     .oneshot((proto.tls, io));
-                return Box::new(drain.watch(fwd.map_err(Into::into), |_| {}));
+                return Box::pin(async move {
+                    tokio_02::pin!(fwd);
+                    drain.watch(fwd.map_err(Into::into), |_| {}).await;
```

Comment:
`Watch::after` was added because it doesn't require the future to be `Unpin`, which is necessary for `Watch::watch` (due to `select` having to borrow the future to poll it).

It could be used here as well. I just didn't think about it. Good call!